app/models/nic/managed.rb

```diff
@@ -128,7 +128,7 @@ def failure(msg, backtrace = nil, dest = :base)
 
     # we must also clone host object so we can detect host attributes changes
     def setup_clone
-      return if new_record?
+      return if new_record? or host.new_record?
```

Comment:
I'm slowly getting to reproduce it. So far I don't think this would be the right solution as discussed on IRC.